DOM_creation/Lesson3/main.js

```diff
@@ -0,0 +1,22 @@
+const ul = document.getElementById("js-ul");
+
+const a1 = document.createElement("a");
+a1.href = "a1.html";
+
+const a2 = document.createElement("a");
+a2.href = "a2.html";
+
+const img1 = document.createElement("img");
+img1.src = "/img/bookmark.png";
+
+const img2 = document.createElement("img");
+img2.src = "/img/message.png";
+
+const array = [{a: a1, img: img1, text: "a1"}, {a: a2, img: img2, text: "a2"}];
+
+array.forEach(e =>{
+  let li = document.createElement('li');
+  e.a.textContent= e.text;
+  li.appendChild(e.a).insertAdjacentElement('afterbegin', e.img);
+  ul.appendChild(li);
```

Comment:
IMO
In this case, there are two `li` so there may not be much impact, but if the number of `li`'s increases to 10, 20, and so on, each time a child element of `ul` is added, the processing may take a long time.
So, how about using [createDocumentFragment()](https://developer.mozilla.org/ja/docs/Web/API/Document/createDocumentFragment)?

Also, I share Monaka's blog post because it is easy to understand.
https://monatamalog.com/archives/291

Reply:
Thank you for your feeback Akira-san,
The link you shared was a really big help for me to grab a better picture of createDocumentFragment() method.